### ISSUE TITLE
Users split into 2 separate zoom calls

### ISSUE THREAD

**Issue by bellesea** [hackclub/slash-z#115]
Call ID: mquj6n
Narrative: I clicked to join a slash z meeting at around 7:58pm on July 19 and found another attendee there. We waited till 8:03 and no one else joined so we DMed others, which revealed that they were all in another zoom call.

Call links:
1: https://hackclub.zoom.us/j/84342679356
2: https://hackclub.zoom.us/j/87217153132

Thanks!

**Comment by JosiasAurel** [comment]
This is what I've from the database logs

*ErrorLogs*

- 2023-07-19T05:56:37.200Z
```
Error: out of open hosts!    at default (file:///app/api/open-zoom-meeting.js:47:11)    at async default (file:///app/api/endpoints/slack/slash-z.js:63:15)    at async file:///app/api/endpoints/slack/index.js:13:9    at async default (file:///app/api/endpoints/slack/index.js:6:10)    at async file:///app/routes.js:46:13
```

- 2023-07-19T14:06:33.833Z
```
Error: out of open hosts!    at default (file:///app/api/open-zoom-meeting.js:47:11)    at async default (file:///app/api/endpoints/slack/slash-z.js:63:15)    at async file:///app/api/endpoints/slack/index.js:13:9    at async default (file:///app/api/endpoints/slack/index.js:6:10)    at async file:///app/routes.js:46:13
```

**Comment by grymmy** [comment]
We have a report that this happened in a meeting today as well - https://hack.af/z-join?id=mepauo

**Comment by JosiasAurel** [comment]
we added tests to find out whether or not two (or more) parallel requests does not lead to different calls and they pass so far see [here](https://github.com/hackclub/slash-z/blob/80903995816db18ba28bc1ad45c60195c6927bdb/race.test.js)

**Comment by grymmy** [comment]
@JosiasAurel I think it also important to note that we added brand-new logic in slash-z to invalidate old zoom URLs in situations such as these to reduce the likelihood individuals expecting to join the same zoom room end up being split.

**Comment by JosiasAurel** [comment]
here is the PR that does that #135

**Comment by maxwofford** [comment]
I think I just experienced this again with scheduled link `4brejs` on Monday Augh 28th at 14:00 eastern tz.

**Comment by grymmy** [comment]
Note - today we had a bad deploy to slash-z which could potentially explain this - however we will investigate fully 
& root cause this, @maxwofford.

**Comment by grymmy** [comment]
zstory output for @maxwofford 's meeting link:

<img width="569" alt="Screenshot 2023-08-28 at 3 20 21 PM" src="https://github.com/hackclub/slash-z/assets/43505167/125c521b-812c-4907-9683-1207a37769e5">

**Comment by grymmy** [comment]
We are able to reproduce this issue in the staging environment now.  Steps to reproduce:

1. Create a scheduled zoom call via gcal.
2. Have Person A join the call
3. Wait >= 2m30s
4. Have Person B join the call

Expected result:  Person B joins the existing call with Person A
Actual result: Person B gets placed into a new call

**Comment by JosiasAurel** [comment]
Slash-z was getting rate-limited by zoom because the /stats endpoint will pull zoom metrics thousands of times every day while the closeZoomCall function relies on valid zoom metrics before deciding to close a call or not. 

closeZoomCall did not explicitly check if we got a 200 (OK) response before marking a call as ended. As a result, it would mark the end date of some calls every time the closeStaleCalls job would run resulting in people finding themselves in different rooms.

This is fixed in #162. #164 will supposedly limit the number of time we query the zoom metrics a day.
